pyomo/contrib/gdpopt/solve_master_problem.py

```diff
@@ -0,0 +1,185 @@
+    """Solves the linear GDP model and attempts to resolve solution issues.
+    Returns one of TerminationCondition.optimal, TerminationCondition.feasible,
+    TerminationCondition.infeasible, or TerminationCondition.unbounded.
+    """
+    timing = solver.timing
+    m = util_block.model()
+
+    if config.mip_presolve:
+        try:
+            # ESJ: deactivating satisfied constraints here is risky since it can
+            # result in sending no constraints to the solver and hence not
+            # actually filling in the values of the variables that got
+            # implicitly fixed by these bounds. We can fix this by calling some
+            # contrib.preprocessing transformations, but for now I'm just
+            # leaving the constraints in.
+            fbbt(m, integer_tol=config.integer_tolerance,
+                 deactivate_satisfied_constraints=False)
+            # [ESJ 1/28/22]: Despite being a little scary, the tightened bounds
+            # are okay to leave in because if you tighten the bounds now, they
+            # could only get tighter in later iterations, since you are
+            # tightening this relaxation
+        except InfeasibleConstraintException:
+            config.logger.debug("MIP preprocessing detected infeasibility.")
+            return tc.infeasible
+
+    # Deactivate extraneous IMPORT/EXPORT suffixes
+    getattr(m, 'ipopt_zL_out', _DoNothing()).deactivate()
+    getattr(m, 'ipopt_zU_out', _DoNothing()).deactivate()
+
+    # Create solver, check availability
+    if not SolverFactory(config.mip_solver).available():
+        raise RuntimeError(
+            "MIP solver %s is not available." % config.mip_solver)
+
+    # Callback immediately before solving MIP master problem
+    config.call_before_master_solve(solver, m, util_block)
+
+    try:
+        with SuppressInfeasibleWarning():
+            mip_args = dict(config.mip_solver_args)
+            elapsed = get_main_elapsed_time(timing)
+            remaining = max(config.time_limit - elapsed, 1)
+            if config.mip_solver == 'gams':
+                mip_args['add_options'] = mip_args.get('add_options', [])
+                mip_args['add_options'].append('option reslim=%s;' % remaining)
+            elif config.mip_solver == 'multisolve':
+                mip_args['time_limit'] = min(mip_args.get('time_limit',
+                                                          float('inf')),
+                                             remaining)
+            results = SolverFactory(config.mip_solver).solve(m, **mip_args)
+
+    except RuntimeError as e:
+        # ESJ: I don't know enough about GAMS to know if this is a good
+        # idea. Does this always mean it's infeasible?
+        if 'GAMS encountered an error during solve.' in str(e):
+            config.logger.warning(
+                "GAMS encountered an error in solve. Treating as infeasible.")
+            return tc.infeasible
```

Comment:
This is making a big assumption that a runtime error in GAMS is only coming from a trivially infeasible problem (there might be other causes, but if we assume perfect translation of the model from GAMS.py, and perfectly working GAMS installation) we can say that the problem is trivially infeasible. At this point, FBBT should've told you that I guess, I would be in favor of changing tc.infeasible to something else. tc.error maybe?

Reply:
OK, thank you for your input here. I think I'll change to tc.error.

Reply:
Actually, on second thought, I'm going to let it raise the error. If we think it really is one, then the try-except is just kicking the can down the road--this will fail when updating the dual bound anyway, if we return tc.error.